core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -560,6 +561,7 @@ class SparkContext(config: SparkConf) extends Logging {
 
     setupAndStartListenerBus()
     postEnvironmentUpdate()
+    _env.metricsSystem.registerSource(appStatusSource)
```

Comment:
Better to put this line to +574

Reply:
Yeah I was wondering where that would be bet to put. So at the end makes sense on the other hand what if I want to have metrics as early as the app starts. Anyway, for now its ok.